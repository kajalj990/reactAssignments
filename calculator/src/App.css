
.result {
  height: 60px;
  background-color: rgb(51, 51, 51);
  width: 100%;
}

.result p {
  font-size: 40px;
  margin: 0px;
  color: white
}

.calculator-body {
  max-width: 200px;
  margin:auto;
  border: 1px solid rgb(128, 111, 111);
}

.button {
  display: block;
}

.colorspe{
  background-color: rgb(194, 35, 75);
}

button {
  width: 25%;
  height: 50px;
  font-size: 20px;
  border: 1px solid red;
}

.clear{
  width:75%
}
