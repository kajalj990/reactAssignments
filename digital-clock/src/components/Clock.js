import React, { Component } from 'react'
import Moment from 'moment'
class Clock extends Component {
    constructor(props) {
        super(props)
        this.state = {
            time: "",
            date: "",
            isHidden: true
        }
    }
    componentWillMount() {
        this.getCurrentTime();
    }
    getCurrentTime = () => {

        let hour = new Date().getHours();
        let min = new Date().getMinutes();
        let sec = new Date().getSeconds();

        this.setState({ time: hour + ':' + min + ':' + sec });
    }
    toggleHidden() {

        let day = new Date()
        this.setState({
            date: day.toDateString(),
            isHidden: !this.state.isHidden

        })
    }
    componentWillUnmount() {
        clearInterval(this.timer);
    }
    componentDidMount() {
        this.timer = setInterval(() => {
            this.getCurrentTime();
        }, 1000);
    }
    render() {
        return (
            <div>
                <label className="switch">
                    <input type="checkbox" onClick={this.toggleHidden.bind(this)}/>
                        <span className="slider round"></span>
                    </label>

                    <div className="design">
                        <h1>{this.state.time}</h1>
                        {!this.state.isHidden && this.state.date}</div>
            </div>
        )
    }

}
export default Clock
