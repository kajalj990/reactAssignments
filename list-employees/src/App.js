import React from "react";
import TableClass from "./components/TableClass";
//import TableClass from "./components/TableClass";
var data = [
  {
    id: 1,
    name: "Kajal",
    Job: "Janitor"
  },
  {
    id: 2,
    name: "Palak",
    Job: "Bouncer"
  },
  {
    id: 3,
    name: "Shivani",
    Job: "Aspiring actress"
  },
  {
    id: 4,
    name: "Pradnya",
    Job: "Bartender"
  }
];
function App() {
  return (
    <div className="App">
      <TableClass data={data} />
    </div>
  );
}

export default App;
