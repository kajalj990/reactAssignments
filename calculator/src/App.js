import React, { Component } from 'react'
import './App.css';
import Result from './components/Result';
import KeyPad from './components/KeyPad'
class App extends Component {
  constructor() {
    super()

    this.state = {
      result: ''
    }
  }

  onClick = button => {
    if (button === '=') {
      this.calculate()
    }
    else if (button === "Backspace") {
      this.backspace()
    }
    else {
      this.setState({
        result: this.state.result + button
      })
    }
  }

  backspace = () => {
    this.setState({
      result: " "
    })
  }

  calculate = () => {
    var ans = ''
    if (this.state.result.includes('--')) {
      ans = this.state.result.replace('--', '+')
    }
    else {
      ans = this.state.result
    }
    this.setState({
      result: (eval(ans) || "") + ""
    })
  }

  render() {
    return (
      <div >
        <h1>Simple Calculator
          </h1>
        <div className="calculator-body">
          
          <Result result={this.state.result} />
          <KeyPad onClick={this.onClick} />
        </div>
      </div>
    )
  }
}

export default App




