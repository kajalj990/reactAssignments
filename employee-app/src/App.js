import React, { Component } from 'react';
import './App.css';
import Form from './components/Form';
import Table from './components/Table';



 class App extends Component {
  state = {
    employees: [],
  };

  submitValueHandler = (name, job, id) => {
    const emp = {
      name: name,
      job: job,
      id: id,
    };
    let arr = [...this.state.employees];
    arr.push(emp);
    this.setState({
      employees: arr,
    });
  };

  deleteButtonHandler = (id) => {
    let index = this.state.employees.findIndex(emp => emp.id === id);
    let empArray = this.state.employees;
    empArray.splice(index, 1);
    this.setState({employees: empArray});
  }

  render() {
    return (
      <div>
        <Table employeeList={this.state.employees} deleteHandle={this.deleteButtonHandler}/>
        <Form changeValue={this.submitValueHandler} />
      </div>
    );
  }
}

export default App
