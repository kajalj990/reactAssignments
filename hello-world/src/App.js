import React from 'react';
//import logo from './logo.svg';
import './App.css';

function App() {

  return (
  <div>
    <h1 id='header'>Hello World</h1>
  </div>);
}

export default App;
