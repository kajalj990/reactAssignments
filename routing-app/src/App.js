import React,{Component} from 'react';

import './App.css';
import {BrowserRouter as Router, NavLink} from 'react-router-dom';
import Route from 'react-router-dom/Route';
class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
        <nav className="ist-group list-group-horizontal">
        <ul>
        <li><NavLink activeStyle={{background:'pink'}} to="/home">Home</NavLink></li>
          
      
          <li><NavLink activeStyle={{background:'pink'}} to="/project">Project</NavLink></li>
     
      
          <li><NavLink activeStyle={{background:'pink'}} to="/service">Service</NavLink></li>
        
       
          <li><NavLink activeStyle={{background:'pink'}} to="/contact">Contact</NavLink></li>
        </ul>
        </nav>
        
          <Route path="/home" render={
            ()=>{
              return (<h1>You chose <span>home</span></h1>);
            }

          }></Route>
          <Route path="/project" render={
            ()=>{
              return (<h1>You chose <span>projects</span></h1>);
            }
          }></Route>
          <Route path="/service" render={
            ()=>{
              return (<h1>You chose <span>service</span></h1>);
            }
          }></Route>
          <Route path="/contact" render={
            ()=>{
              return (<h1>You chose <span>contact</span></h1>);
            }
          }></Route>
          
        </div>
      </Router>
    )
  }
}

export default App

