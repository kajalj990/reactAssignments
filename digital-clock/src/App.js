import React from 'react'
import './App.css';
import Clock from './components/Clock'
function App() {
  return (
    <div className="App">
      <h1 className="App-header"> Digital Clock</h1>
      <Clock className = 'clock'/>
    </div>
  )
}

export default App
