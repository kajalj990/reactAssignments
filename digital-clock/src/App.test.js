import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { shallow } from 'enzyme'
import Clock from './components/Clock';
import renderer from 'react-test-renderer';
test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText('Digital Clock');
  expect(linkElement).toBeInTheDocument();
});

test('Snapshot',()=>{
  const clock= shallow(<Clock/>);
  const tree = renderer.create(<Clock/>).toJSON();
 // clock.setState({time:new Date().getTime()});
  expect(tree).toMatchSnapshot();
})

test('State Changes successfully',()=>{
  const clock= shallow(<Clock/>);
 // const tree = renderer.create(<Clock/>).toJSON();
  clock.setState({time:new Date().getTime()});
  expect(clock).toMatchSnapshot();
})
//time zone 
//toggle button
//date 