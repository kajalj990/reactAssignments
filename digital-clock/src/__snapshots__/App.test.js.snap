// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should change state 1`] = `
<div>
  <label
    className="switch"
  >
    <input
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
  <div
    className="design"
  >
    <h1>
      9:55:43
    </h1>
  </div>
</div>
`;

exports[`Should match 1`] = `ShallowWrapper {}`;

exports[`Snapshot 1`] = `
<div>
  <label
    className="switch"
  >
    <input
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
  <div
    className="design"
  >
    <h1>
      9:57:46
    </h1>
  </div>
</div>
`;

exports[`State Changes successfully 1`] = `ShallowWrapper {}`;
